src/Evolution/Systems/ScalarWave/Characteristics.hpp

```diff
@@ -0,0 +1,226 @@
+ * \f}
+ *
+ * The characteristic fields \f$\hat{u}_\alpha\f$ are given in terms of the
+ * evolved fields by Eq.(33) - (35) of \cite Holst2004wt, respectively:
+ * \f{align*}
+ * u^{\psi} =& \psi \\
+ * u^0_{i} =& (\delta^k_i - n_i n^k) \Phi_{k} := P^k_i \Phi_{k} \\
+ * u^{\pm} =& \Pi \pm n^i \Phi_{i} - \gamma_2\psi
+ * \f}
+ *
+ * where \f$\psi\f$ is the scalar field, \f$\Phi_{i}=\partial_i \psi\f$ is an
+ * auxiliary variable, \f$\Pi\f$ is a conjugate momentum, \f$\gamma_2\f$
+ * is a constraint damping parameter, and \f$n_k\f$ is the unit normal to the
+ * surface along which the characteristic fields are defined.
+ *
+ * \ref EvolvedFieldsFromCharacteristicFieldsCompute computes evolved fields
+ * \f$u_\alpha\f$ in terms of the characteristic fields. This uses the inverse
+ * of above relations:
+ *
+ * \f{align*}
+ * \psi =& u^{\psi}, \\
+ * \Pi =& \frac{1}{2}(u^{+} + u^{-}) + \gamma_2 u^{\psi}, \\
+ * \Phi_{i} =& \frac{1}{2}(u^{+} - u^{-}) n_i + u^0_{i}.
+ * \f}
+ *
+ * The corresponding characteristic speeds \f$\lambda_\alpha\f$ are computed by
+ * \ref CharacteristicSpeedsCompute .
+ */
+template <size_t Dim>
+Variables<tmpl::list<Tags::UPsi, Tags::UZero<Dim>, Tags::UPlus, Tags::UMinus>>
+characteristic_fields(const Scalar<DataVector>& gamma_2,
+                      const Scalar<DataVector>& psi,
+                      const Scalar<DataVector>& pi,
+                      const tnsr::i<DataVector, Dim, Frame::Inertial>& phi,
+                      const tnsr::i<DataVector, Dim, Frame::Inertial>&
+                          unit_normal_one_form) noexcept;
+
+template <size_t Dim>
+void characteristic_fields(
+    gsl::not_null<Variables<
+        tmpl::list<Tags::UPsi, Tags::UZero<Dim>, Tags::UPlus, Tags::UMinus>>*>
+        char_fields,
+    const Scalar<DataVector>& gamma_2, const Scalar<DataVector>& psi,
+    const Scalar<DataVector>& pi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& phi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        unit_normal_one_form) noexcept;
+
+template <size_t Dim>
+struct CharacteristicFieldsCompute : Tags::CharacteristicFields<Dim>,
+                                     db::ComputeTag {
+  using base = Tags::CharacteristicFields<Dim>;
+  using type = typename base::type;
+  using argument_tags =
+      tmpl::list<Tags::ConstraintGamma2, Psi, Pi, Phi<Dim>,
+                 ::Tags::Normalized<::Tags::UnnormalizedFaceNormal<Dim>>>;
+
+  static typename Tags::CharacteristicFields<Dim>::type function(
+      const Scalar<DataVector>& gamma_2, const Scalar<DataVector>& psi,
+      const Scalar<DataVector>& pi,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& phi,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>&
+          unit_normal_one_form) noexcept {
+    return characteristic_fields(gamma_2, psi, pi, phi, unit_normal_one_form);
+  };
+};
+// @}
+
+// @{
+/*!
+ * \ingroup ScalarWave
+ * \brief For expressions used here to compute evolved fields from
```

Comment:
The brief should still say what this does. I suggest, to match the other functions, adding a brief that says, `Computes evolved fields from characteristic fields`,
and keeping the current sentence (which is good) as the main documentation below the brief.

Reply:
done